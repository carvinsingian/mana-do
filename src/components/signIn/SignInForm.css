.formContainer {
  height: 100%;
  width: 100%;
  color: #e5e5e5;
  max-width: 500px;
  margin: auto;
  background-color: rgba(255, 255, 255, 0.13);
  padding: 60px;
  border-radius: 8px;
}

.formContainer > h1 {
  text-align: center;
  margin-bottom: 24px;
}

.formItem {
  display: flex;
  flex-direction: column;
  margin-top: 16px;
}

.formItemLabel {
  margin-bottom: 4px;
  font-weight: 500;
}

.formError {
  margin-top: 4px;
  color: red;
  font-size: 0.75rem;
  min-height: 18px;
}

.formButton {
  margin-top: 24px;
}
