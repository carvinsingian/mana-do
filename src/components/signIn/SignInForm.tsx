import React, { useState } from 'react'

import {useHistory} from 'react-router-dom'
import Service from '../../service'
import Button from '../../ui/button/Button'

import './SignInForm.css'



const SignInForm = () => {
  const [form, setForm] = useState({
    userId: '',
    password: ''
  })
  //Adding error handling for form
  const [error, setError] = useState<string>(' ')
  const history = useHistory()

  const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    setError('')
    Service.signIn(form.userId, form.password)
      .then(resp => {
        localStorage.setItem('token', resp)
        history.push('/todo')
      })
      .catch(error => {
        setError(error)
      })
  }

  const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
      e.persist()
      setError('')
      setForm(prev=>({
          ...prev,
          [e.target.name]: e.target.value
      }))
  }

  return (
    <div className="formContainer">
      <h1>Mana Do Application</h1>
      <form onSubmit={signIn} data-testid="sign-in-form">
        <div className="formItem">
          <label htmlFor="user_id" className="formItemLabel">
            Username
          </label>
          <input
            id="user_id"
            data-testid="sign-in-input-userid"
            name="userId"
            required
            value={form.userId}
            onChange={onChangeField}
            placeholder="Type in your username"
          />
        </div>
        <div className="formItem">
          <label htmlFor="password" className="formItemLabel">
            Password
          </label>
          <input
            id="password"
            data-testid="sign-in-input-password"
            name="password"
            type="password"
            required
            value={form.password}
            onChange={onChangeField}
            placeholder="Type in your password"
          />
        </div>
        <div>
          <p className="formError">{error}</p>
        </div>
        <Button 
          className="formButton" 
          type="submit"
          dataTestId="sign-in-button"
        >
          Log In
        </Button>
      </form>
    </div>
  )
}


export default SignInForm