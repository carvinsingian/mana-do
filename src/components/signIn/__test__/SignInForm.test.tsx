import React from 'react'
import ReactDOM from 'react-dom'
import SignInForm from '../SignInForm'
import { render, cleanup } from '@testing-library/react'

afterEach(cleanup)

test('Sign in form renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(<SignInForm />, div)
})

test('Sign in page matches snapshot', () => {
  const { asFragment } = render(<SignInForm/>)

  const componentRender = asFragment()
  expect(componentRender).toMatchSnapshot()
})

test('Sign in form renders the form', () => {
  const { getByTestId } = render(<SignInForm/>)
  const form = getByTestId('sign-in-form')

  expect(form).toBeInTheDocument()
})

test('Sign in form has username field', () => {
  const { getByTestId } = render(<SignInForm/>)
  const username = getByTestId('sign-in-input-userid')

  expect(username).toBeInTheDocument()
  expect(username).toBeRequired()
})

test('Sign in form has password field', () => {
  const { getByTestId } = render(<SignInForm/>)
  const password = getByTestId('sign-in-input-password')

  expect(password).toBeInTheDocument()
  expect(password).toBeRequired()
})

test('Sign in form has sign in button', () => {
  const { getByTestId } = render(<SignInForm/>)
  const button = getByTestId('sign-in-button')

  expect(button).toBeInTheDocument()
})


