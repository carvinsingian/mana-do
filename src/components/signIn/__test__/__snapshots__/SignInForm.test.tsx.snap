// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="formContainer"
  >
    <h1>
      Mana Do Application
    </h1>
    <form
      data-testid="sign-in-form"
    >
      <div
        class="formItem"
      >
        <label
          class="formItemLabel"
          for="user_id"
        >
          Username
        </label>
        <input
          data-testid="sign-in-input-userid"
          id="user_id"
          name="userId"
          placeholder="Type in your username"
          required=""
          value=""
        />
      </div>
      <div
        class="formItem"
      >
        <label
          class="formItemLabel"
          for="password"
        >
          Password
        </label>
        <input
          data-testid="sign-in-input-password"
          id="password"
          name="password"
          placeholder="Type in your password"
          required=""
          type="password"
          value=""
        />
      </div>
      <div>
        <p
          class="formError"
        >
           
        </p>
      </div>
      <button
        class="formButton button button__primary"
        data-testid="sign-in-button"
        type="submit"
      >
        Log In
      </button>
    </form>
  </div>
</DocumentFragment>
`;
